curvefs/src/client/s3/disk_cache_base.cpp

```diff
@@ -133,23 +138,36 @@ int DiskCacheBase::LoadAllCacheFile(std::set<std::string> *cachedObj) {
                                std::set<std::string> *cacheObj) -> bool {
         DIR *dir;
         struct dirent *ent;
-        std::string fileName, nextdir;
-        if ((dir = posixWrapper_->opendir(path.c_str())) != NULL) {
-            while ((ent = posixWrapper_->readdir(dir)) != NULL) {
+        dir = posixWrapper_->opendir(path.c_str());
+        if (dir != nullptr) {
+            while ((ent = posixWrapper_->readdir(dir)) != nullptr) {
```

Comment:
```suggestion
        Dir* dir = posixWrapper_->opendir(path.c_str());
        if (dir != nullptr) {
            struct dirent *ent;
            while ((ent = posixWrapper_->readdir(dir)) != nullptr) {
```

Reply:
it's better to handle `dir == nullptr` firstly.

Reply:
> it's better to handle `dir == nullptr` firstly.

fix